### DISCUSSION TITLE
Using custom constraint / index names not work?

### DISCUSSION THREAD

**Discussion by 45592858** [prisma/prisma#14756] [Q&A] [answered]
When I migrate from 2.17 to 3.15, run "npx prisma migrate dev", it since not work on "Using custom constraint / index names" ;
And also found "Removed foreign / Added foreign key" problem.
problem see attachment picture
![ff](https://user-images.githubusercontent.com/17882261/184083616-63ce8902-7254-41e0-b3a5-6f848153c2f9.png)
.

**Comment by janpio** [comment]
Please have a look at https://github.com/prisma/prisma/releases/tag/3.0.1, there were some changes to this in 3.x that might need manual actions from you.

**Comment by 45592858** [comment, marked as answer]
Thanks, problem solved. 1) discarded "migrations history" of before, 2)"prisma db pull" re-gen schema, 3)"prisma migrate dev" re-init, 4) then apply new schema change, it worked.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
